module/zfs/zfs_fm.c

```diff
@@ -131,7 +131,8 @@ zfs_ereport_start(nvlist_t **ereport_out, nvlist_t **detector_out,
 	 * If we are doing a spa_tryimport() or in recovery mode,
 	 * ignore errors.
 	 */
-	if (spa_load_state(spa) == SPA_LOAD_TRYIMPORT ||
+	if (spa != NULL ||
+	    spa_load_state(spa) == SPA_LOAD_TRYIMPORT ||
```

Comment:
This chunk isn't correct.  Yes it prevents a NULL dereference in this block but the same thing can happen 5 lines down.  Please drop think hunk from the patch and we'll address it if need be in a different PR.